event_session/models/event_session.py

```diff
@@ -0,0 +1,275 @@
+    )
+    registration_ids = fields.One2many(
+        comodel_name='event.registration',
+        inverse_name='session_id',
+        string='Attendees',
+        state={'done': [('readonly', True)]},
+    )
+    event_mail_ids = fields.One2many(
+        comodel_name='event.mail',
+        inverse_name='session_id',
+        string='Mail Schedule',
+        copy=True
+    )
+
+    @api.multi
+    @api.depends('date_begin', 'date_end')
+    def _compute_name(self):
+        setlocale(LC_ALL, locale=(self.env.lang, 'UTF-8'))
+        for session in self:
+            if not (session.date_begin and session.date_end):
+                session.name = '/'
+                continue
+            date_begin = fields.Datetime.from_string(
+                session.date_begin_located)
+            date_end = fields.Datetime.from_string(session.date_end_located)
+            dt_format = '%A %d/%m/%y %H:%M'
+            name = date_begin.strftime(dt_format)
+            if date_begin.date() == date_end.date():
+                dt_format = '%H:%M'
+            name += " - " + date_end.strftime(dt_format)
+            session.name = name.capitalize()
+
+    def _session_mails_from_template(self, event_id, mail_template=None):
+        vals = [(6, 0, [])]
+        if not mail_template:
+            mail_template = self.env['ir.values'].get_default(
+                'event.config.settings', 'event_mail_template_id')
+            if not mail_template:
+                # Not template scheduler defined in event settings
+                return vals
+        if isinstance(mail_template, int):
+            mail_template = self.env['event.mail.template'].browse(
+                mail_template)
+        for scheduler in mail_template.scheduler_template_ids:
+            vals.append((0, 0, {
+                'event_id': event_id,
+                'interval_nbr': scheduler.interval_nbr,
+                'interval_unit': scheduler.interval_unit,
+                'interval_type': scheduler.interval_type,
+                'template_id': scheduler.template_id.id,
+            }))
+        return vals
+
+    @api.multi
+    def name_get(self):
+        """Redefine the name_get method to show the event name with the event
+        session.
+        """
+        res = []
+        for item in self:
+            res.append((item.id, "[%s] %s" % (item.event_id.name, item.name)))
+        return res
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('event_mail_ids', False):
+            vals.update({
+                'event_mail_ids':
+                    self._session_mails_from_template(vals['event_id'])
+            })
+        return super(EventSession, self).create(vals)
+
+    @api.multi
+    @api.depends('seats_max', 'registration_ids.state')
+    def _compute_seats(self):
+        """Determine reserved, available, reserved but unconfirmed and used
+        seats by session.
+        """
+        # initialize fields to 0
+        for session in self:
```

Comment:
This is not needed, as by default all fields are reset to their default values (0 in this case).